server/service/osquery_utils/queries.go

```diff
@@ -1187,17 +1187,25 @@ func GetDetailQueries(fleetConfig config.FleetConfig, features *fleet.Features)
 		generatedMap["scheduled_query_stats"] = scheduledQueryStats
 	}
 
-	for _, env := range os.Environ() {
-		prefix := "FLEET_DANGEROUS_REPLACE_"
-		if !strings.HasPrefix(env, prefix) {
-			continue
+	if features != nil {
+		var unknownQueries []string
+
+		for name, override := range features.DetailQueryOverrides {
+			if override == nil {
+				delete(generatedMap, name)
+			} else {
+				query, ok := generatedMap[name]
+				if !ok {
+					unknownQueries = append(unknownQueries, name)
+					continue
+				}
+				query.Query = *override
+				generatedMap[name] = query
+			}
 		}
-		if i := strings.Index(env, "="); i >= 0 {
-			queryName := strings.ToLower(strings.TrimPrefix(env[:i], prefix))
-			newQuery := env[i+1:]
-			query := generatedMap[queryName]
-			query.Query = newQuery
-			generatedMap[queryName] = query
+
+		if len(unknownQueries) > 0 {
+			logging.WithExtras(ctx, "unknown-queries", strings.Join(unknownQueries, ","))
```

Comment:
This may be logged as debug only, see: https://github.com/fleetdm/fleet/blob/2bc21a048686d7ccfd03392592d132b61a62a513/server/contexts/logging/logging.go#L116-L123

That's why on some places we call `logging.WithExtras` + `logging.WithErr`:
https://github.com/fleetdm/fleet/blob/2bc21a048686d7ccfd03392592d132b61a62a513/server/service/osquery.go#L835-L836

Reply:
Hm, yeah, so lets treat this as an error.